### PR TITLE
EVG-6384: add required methods for build view in Spruce

### PR DESCRIPTION
**Pull request by dominoweir**
first off- I apologize in advance for how many lines this says I've changed. The spacing and indentation has been wonky in a few of these files for a while now, but it only just got to the point where I was annoyed enough to go ahead and change it. Besides that, this set of changes includes adding the getLogs, getTasks, getTests, and getBuild methods, some tests for those methods, and new models required for the build view page.

### CODE REVIEW

File: src/evergreen.ts [hunk 1]
```diff
@@ -1,235 +1,262 @@
...
-        headers: headers,
-        url: url,
+  /**
+   * Gets logs of a particular type for a particular task/
+   * @param callback - function to process the response
+   * @param id - identifier for the task whose logs we want
+   * @param type - type of log to return
+   * @returns nothing
+   */
+  public getLogs(callback: request.RequestCallback, url: string) {
```
Comment (john-m-liu) [inline src/evergreen.ts]: We shouldn't require the caller to know the URL. The function signature before was fine - would it work to just append the text=true?

File: src/evergreen.ts [hunk 2]
```diff
@@ -1,235 +1,262 @@
...
+    const resource = "builds/" + id;
+    this.getAPIResource(callback, apiV2Resource(resource));
+  }
+
+  /**
+   * Gets tasks for a given build ID
+   * @param id - build ID whose tasks we want
+   * @returns nothing
+   */
+  public getTasks(callback: request.RequestCallback, id: string) {
```
Comment (john-m-liu) [inline src/evergreen.ts]: call this getTasksForBuild and name the parameter buildID (to avoid ambiguity with a task ID)

File: src/evergreen.ts [hunk 3]
```diff
@@ -1,235 +1,262 @@
...
+    const resource = "builds/" + id + "/tasks";
+    this.getAPIResource(callback, apiV2Resource(resource));
+  }
+
+  /**
+   * Gets tests for a given task ID
+   * @param id - task ID whose tests we want
+   * @returns nothing
+   */
+  public getTests(callback: request.RequestCallback, id: string) {
```
Comment (john-m-liu) [inline src/evergreen.ts]: call this getTestsForTask and name the parameter taskID

File: src/evergreen.ts (hunk 1 above)
Comment (dominoweir) [reply src/evergreen.ts]: Yes, that works. I was messing around to see if using the URLs passed in the task response would work in this method, which they do, but I agree it makes more sense to not pass the entire URL.

File: src/evergreen.ts (hunk 2 above)
Comment (dominoweir) [reply src/evergreen.ts]: Fixed

File: src/evergreen.ts (hunk 3 above)
Comment (dominoweir) [reply src/evergreen.ts]: Fixed

**Pull request merged**